### PR TITLE
BF: MonitorCenter and ColorCal

### PR DESCRIPTION
**Pull request by schofiaj**
Fixes multiple problems with MonitorCenter.py calling obsolete functions and colorcal.py not converting str to byte and vice versa.

### CODE REVIEW

**Comment by hoechenberger** [comment]
Hi, are the changes in `colorcal.py` compatible with both Python 2 and 3?

**Comment by schofiaj** [comment]
Testes with PY3 32bit, PY3 64 bit and PY2

File: psychopy/hardware/crs/colorcal.py [hunk 1]
```diff
@@ -143,17 +141,21 @@ def sendMessage(self, message, timeout=0.1):
...
-            if thisLine in (eol, '>', ''):  # lines we don't care about
+            if thisLine in (eol.encode('ascii'), '>'.encode('ascii'), ''.encode('ascii')):  # lines we don't care about
                 nEmpty += 1
                 continue
             else:
                 # line without any eol chars
-                lines.append(thisLine.rstrip(eol))
+                #myLine=thisLine.encode('ascii')
+                #myLine=thisLine.rstrip(eol.encode('ascii'))
+                #lines.append(thisLine)
```
Comment (hoechenberger) [inline psychopy/hardware/crs/colorcal.py]: I guess these are leftovers from testing and should be removed?

File: psychopy/hardware/crs/colorcal.py [hunk 2]
```diff
@@ -206,13 +209,16 @@ def getInfo(self):
...
+        val = self.sendMessage(b'IDR')
+        valstrip=val.strip(b'\n\r>')
+        val=valstrip.split(b',')
+        ok = (val[0] == b'OK00')
         if ok:
             firmware = val[2]
             serialNum = val[4]
             firmBuild = val[-1]
         else:
+            print('not ok')
```
Comment (hoechenberger) [inline psychopy/hardware/crs/colorcal.py]: This is probably a testing artifact and should be removed?

File: psychopy/hardware/crs/colorcal.py [hunk 3]
```diff
@@ -206,13 +209,16 @@ def getInfo(self):
         Other values will be a string or None.
 
         """
-        val = self.sendMessage(b'IDR').split(',')
-        ok = (val[0] == 'OK00')
+        val = self.sendMessage(b'IDR')
+        valstrip=val.strip(b'\n\r>')
```
Comment (hoechenberger) [inline psychopy/hardware/crs/colorcal.py]: Could you add spaces around the `=` please?

File: psychopy/hardware/crs/colorcal.py [hunk 4]
```diff
@@ -206,13 +209,16 @@ def getInfo(self):
...
+        val=valstrip.split(b',')
```
Comment (hoechenberger) [inline psychopy/hardware/crs/colorcal.py]: Again, spaces around the `=` please?

File: psychopy/hardware/crs/colorcal.py [hunk 5]
```diff
@@ -282,9 +289,10 @@ def getCalibMatrix(self):
         # parsing?
         for rowN in range(3):
             rowName = 'r0%i' % (rowN + 1)
-            val = self.sendMessage(rowName, timeout=1.0)
-            vals = val.split(',')  # convert to list of values
-            if vals[0] == 'OK00' and len(vals) > 1:
+            val = self.sendMessage(rowName.encode('ascii'), timeout=1.0)
+            valstrip=val.strip(b'\n\r>')
```
Comment (hoechenberger) [inline psychopy/hardware/crs/colorcal.py]: Spaces around `=` please :)

**Comment by hoechenberger** [comment]
Great, thanks!

**Comment by djmannion** [comment]
Thanks for this! I noticed the colorcal issues the other day, great to see them fixed.

**Comment by peircej** [comment]
Great work y'all (especially Andrew!) 😄

**Comment by peircej** [comment]
This pull request **introduces 1 alert** when merging d49f154a0e3a59494af74ab1c67d2b7d777a80fc into 881e33ccc9abd4e9446bda6035848b5bd63b1416 - [view on LGTM.com](https://lgtm.com/projects/g/psychopy/psychopy/rev/pr-7c6227132f2671addbbde73fab54fe445a3d654c)

**new alerts:**

* 1 for Module is imported more than once

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**